src/search/srde.c

```
@@ -0,0 +1,274 @@
+  do {
+    *b = 1 + (int)(sr_rng_uniform01(rng) * pop);
+  } while (*b == skip || *b == *a);
+
+  do {
+    *c = 1 + (int)(sr_rng_uniform01(rng) * pop);
+  } while (*c == skip || *c == *a || *c == *b);
+
+  return 0;
+}
+
+static int sr_de_init_population(sr_rng *rng, int pop, int ndim, real span,
```

Clang (14.0.6)

[warning] 3 adjacent parameters of 'sr_de_init_population' of similar type ('int') are easily swapped by mistake

(bugprone-easily-swappable-parameters)